Simplenote/Classes/SPAuthViewController.swift

```diff
@@ -357,29 +411,47 @@ private extension SPAuthViewController {
         performSimperiumAuthentication()
     }
     
-    @IBAction func performLogInWithMagicLink() {
+    @IBAction func requestLogInCode() {
         guard ensureWarningsAreOnScreenWhenNeeded() else {
             return
         }
-
-        presentPasswordInterface()
         
-// TODO: Restore Mail + Code Auth Flow
-//        lockdownInterface()
-//
-//        let email = self.email
-//        controller.requestLoginEmail(username: email) { error in
-//            // TODO: 429 (Rate Limited)? push PW auth instead
-//            self.presentPasswordInterface()
-//            
-//            if let error {
-//                self.handleError(error: error)
-//            } else {
-//                SPTracker.trackUserRequestedLoginLink()
-//            }
-//            
-//            self.unlockInterface()
-//        }
+        lockdownInterface()
+
+        controller.requestLoginEmail(username: email) { error in
+// TODO: 429 (Rate Limited)? push PW auth instead
+
+            if let error {
+                self.handleError(error: error)
+            } else {
+                self.presentCodeInterface()
+                SPTracker.trackLoginLinkRequested()
+            }
+
+            self.unlockInterface()
+        }
+    }
+    
+    @IBAction func performLogInWithCode() {
+        guard ensureWarningsAreOnScreenWhenNeeded() else {
+            return
+        }
+
+        Task { @MainActor in
+            lockdownInterface()
+            
+            do {
+                try await controller.loginWithCode(username: state.username, code: state.code)
+                SPTracker.trackLoginLinkConfirmationSuccess()
+            } catch let error as SPAuthError {
+                SPTracker.trackLoginLinkConfirmationFailure()
+                self.handleError(error: error)
+            } catch {
+// TODO: Fixme
```

Comment:
To be revisited in a follow up:

1. I'm not fond of not having typed errors
2. I think rate limited responses should be handled in a specific way (ie. pushing the Password UI)